src/components/PostDetails.tsx

```diff
@@ -1,106 +1,165 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { getData } from '../api/fetch';
+import { Comment, CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+
+interface Props {
+  posts: Post[];
+  selectedPost: Post | null;
+}
+
+export const PostDetails: React.FC<Props> = ({ posts, selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [showForm, setShowForm] = useState(false);
+
+  const [loading, setLoading] = useState(false);
+  const [loadingError, setLoadingError] = useState(false);
+
+  useEffect(() => {
+    if (posts && selectedPost) {
+      const fetchComments = async () => {
+        try {
+          const url = `/comments?postId=${selectedPost.id}`;
+
+          setLoading(true);
+
+          if (selectedPost) {
+            setLoadingError(false);
+
+            const currentComments = await getData<Comment>(url);
+
+            setComments(currentComments);
+          }
+        } catch {
+          setLoadingError(true);
+        } finally {
+          setLoading(false);
+        }
+      };
+
+      fetchComments();
+    }
+  }, [posts, selectedPost]);
+
+  useEffect(() => {
+    setShowForm(false);
+  }, [selectedPost]);
+
+  const handleShowForm = () => {
+    setShowForm(true);
+  };
+
+  const handleDelete = async (id: number) => {
+    const updatedCommnets = comments.filter(comment => comment.id !== id);
```

Comment:
There is a typo in the variable name `updatedCommnets`. It should be corrected to `updatedComments` to reflect its purpose accurately.